### PR TITLE
fix: allow resources surrogates

### PR DESCRIPTION
**Pull request by seia-soto**
refs https://github.com/gorhill/uBlock/wiki/Resources-Library
refs https://github.com/ghostery/adblocker/issues/4392
fixes https://github.com/ghostery/adblocker/issues/4393

### CODE REVIEW

File: packages/adblocker/test/resources.test.ts [hunk 1]
```diff
@@ -272,6 +280,12 @@ describe('#Resources', function () {
     it('includes setup for scritplet globals', function () {
       expect(resources.getScriptlet('a')).to.include('var scriptletGlobals = {};');
     });
+
+    it('allows resources surrogate', function () {
+      expect(resources.getScriptlet('surrogate'))
+        .to.include('resource')
+        .not.to.include('scriptletGlobals');
```
Comment (chrmod) [inline packages/adblocker/test/resources.test.ts:287]: please add a comment on what does this express

File: packages/adblocker/src/resources.ts [hunk 2]
```diff
@@ -333,6 +333,24 @@ export default class Resources {
     return script;
   }
 
+  private getResourceAsScriptlet(name: string): Scriptlet | undefined {
```
Comment (chrmod) [inline packages/adblocker/src/resources.ts:336]: if this is private and the the only user need only the body then return a body only

File: packages/adblocker/src/resources.ts [hunk 3]
```diff
@@ -333,6 +333,24 @@ export default class Resources {
...
+    if (name.endsWith('.js') === true) {
+      return undefined;
+    }
+
+    const resource = this.resourcesByName.get(name + '.js');
```
Comment (chrmod) [inline packages/adblocker/src/resources.ts:341]:
```suggestion
    const resource = this.resourcesByName.get(name.endsWith('.js') ? name : `${name}.js`);
```

File: packages/adblocker/src/resources.ts [hunk 4]
```diff
@@ -314,7 +314,7 @@ export default class Resources {
     const scriptlet = this.getRawScriptlet(name);
```
Comment (chrmod) [inline packages/adblocker/src/resources.ts:314]: imao this previous approach was quite fine

```suggestion
    const scriptlet = this.getRawScriptlet(name) || this.getResourceAsScriptlet(name);
```

File: packages/adblocker/test/resources.test.ts [hunk 5]
```diff
@@ -272,6 +280,12 @@ describe('#Resources', function () {
     it('includes setup for scritplet globals', function () {
       expect(resources.getScriptlet('a')).to.include('var scriptletGlobals = {};');
     });
+
+    it('allows resources surrogate', function () {
+      expect(resources.getScriptlet('surrogate'))
```
Comment (chrmod) [inline packages/adblocker/test/resources.test.ts:285]: are those things always called without `.js` ?

**Review by chrmod — CHANGES_REQUESTED** [review]

File: packages/adblocker/test/resources.test.ts (hunk 5 above)
Comment (seia-soto) [reply packages/adblocker/test/resources.test.ts:285]: Yes, resources surrogates are always called without suffix or file extension. Also, uBo gradually plan to drop calling scriptlets with suffix or file extension in the future.

File: packages/adblocker/test/resources.test.ts [hunk 6]
```diff
@@ -272,6 +280,12 @@ describe('#Resources', function () {
     it('includes setup for scritplet globals', function () {
       expect(resources.getScriptlet('a')).to.include('var scriptletGlobals = {};');
     });
+
+    it('allows resources surrogate', function () {
+      // Resource surrogates should be called without `.js` suffix or file extension.
+      // Also, they don't need to be wrapped with scriptlet suppliments like `scriptletGlobals` to avoid possible conflicts.
+      expect(resources.getScriptlet('surrogate')).to.be.equal('(function resource() {})()');
```
Comment (chrmod) [inline packages/adblocker/test/resources.test.ts:287]: please add the expectation for lack of scriptletGlobals

**Review by chrmod — APPROVED** [review]

**Pull request merged**